app/src/js/components/oauth.js

```diff
@@ -1,53 +1,66 @@
-import React, { useState, useEffect } from 'react';
-import { connect } from 'react-redux';
-import withQueryParams from 'react-router-query-params';
-import { withRouter } from 'react-router-dom';
-import PropTypes from 'prop-types';
-import Modal from 'react-bootstrap/Modal';
-import get from 'lodash/get';
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import { Helmet } from 'react-helmet';
+import Modal from 'react-bootstrap/Modal';
+import PropTypes from 'prop-types';
+// import get from 'lodash/get';
 import { login, setTokenState } from '../actions';
 import { window } from '../utils/browser';
 import { buildRedirectUrl } from '../utils/format';
 import _config from '../config';
 import ErrorReport from './Errors/report';
 import Header from './Header/header';
-import { historyPushWithQueryParams } from '../utils/url-helper';
+import { withUrlHelper } from '../withUrlHelper';
 
 const { updateDelay, apiRoot, oauthMethod } = _config;
 
-const OAuth = ({
-  dispatch,
-  api,
-  location,
-  apiVersion,
-  queryParams,
-}) => {
+const OAuth = ({ urlHelper }) => {
   const [token, setToken] = useState(null);
 
-  useEffect(() => {
-    if (api.authenticated) {
-      dispatch(setTokenState(token));
-      const { pathname } = location;
-      if (pathname !== '/auth' && get(window, 'location.reload')) {
-        setTimeout(() => window.location.reload(), updateDelay);
-      } else if (pathname === '/auth') {
-        setTimeout(() => historyPushWithQueryParams('/'), updateDelay); // react isn't seeing this a function
-      }
-    }
-  }, [api.authenticated, dispatch, location, token]);
+  const dispatch = useDispatch();
+  const { location, navigate } = urlHelper;
+
+  const authenticated = useSelector((state) => state.api.authenicated);
+  const api = useSelector((state) => state.api);
+  const apiVersion = useSelector((state) => state.apiVersion);
 
   useEffect(() => {
-    const { token: queryToken } = queryParams;
+    const params = new URLSearchParams(location.search);
+    const queryToken = params.get('token');
+    const stateParam = params.get('state');
+
     if (queryToken) {
       setToken(queryToken);
+
+      let state = {};
+      try {
+        state = JSON.parse(decodeURIComponent(stateParam));
+        console.log('Parsed state:', state);
+      } catch (error) {
+        console.error('Error parsing state:', error);
```

Comment:
What would cause an error parsing the state? An invalid param? If that does happen, should the user be presented with an error?

Reply:
Yes, if the state parameter contains an invalid URL or param. I would think so but that would be another mechanism outside of console debugging.